neat_pvd.c

```diff
@@ -0,0 +1,531 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#ifdef __linux__
+    #include <net/if.h>
+#endif
+
+#include "neat.h"
+#include "neat_internal.h"
+#include "neat_resolver.h"
+#include "neat_core.h"
+#include "neat_pvd.h"
+#include "neat_addr.h"
+
+char *
+compute_reverse_ip(struct neat_addr *src_addr)
+{
+    struct in_addr src_addr4;
+    struct in6_addr src_addr6;
+    char reverse_ip[80]; // maximum length for a reverse /128 IPv6
```

Comment:
Is this sufficient to store the maximum length of the reverse address plus an additional 0-byte (which is added by sprintf) at the end?